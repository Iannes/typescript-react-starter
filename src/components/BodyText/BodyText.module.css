.StepTitle {
  width: 275px;
  width: 240px;
  margin: 20px auto 5px;
  color: #1c1c1c;
  font-size: 20px;
  font-weight: 600;
  line-height: 26px;
  text-align: center;
}
