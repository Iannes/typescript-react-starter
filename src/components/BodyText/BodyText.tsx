import React from "react";
import Typography from "@material-ui/core/Typography";
import { TextProps } from "types";

export const BodyText: React.FC<TextProps> = ({ titleText, variant }) => {
  if (variant === "heading") {
    return (
      <Typography variant="h5" component="h5">
        {titleText}
      </Typography>
    );
  } else {
    return (
      <Typography variant="body1" component="p">
        {titleText}
      </Typography>
    );
  }
};
