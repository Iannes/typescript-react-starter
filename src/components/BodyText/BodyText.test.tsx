import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import BodyText from "./";

beforeEach(() => {
  jest.resetModules();
});

describe("<BodyText />", () => {
  const wrapper = shallow(<BodyText variant="heading" titleText="Heading" />);
  describe("Receives variant and titletext props", () => {
    test("Renders paragraph element by default", () => {
      const wrapper = shallow(<BodyText titleText={"Paragraph t"} />);
    });
    test("If variant prop is heading, it renders element with variant h5", () => {
      expect(wrapper.props().variant).toEqual("h5");
    });
    test("titleText renders the word Heading", () => {
      expect(wrapper.text()).toEqual("Heading");
    });

    test("Renders properly", () => {
      const component = renderer.create(<BodyText titleText={"Paragraph t"} />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
