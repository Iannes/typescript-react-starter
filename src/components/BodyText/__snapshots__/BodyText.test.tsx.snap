// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BodyText /> Receives variant and titletext props Renders properly 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1"
>
  Paragraph t
</p>
`;
